### ISSUE TITLE
Fails to build sway

### ISSUE THREAD

**Issue by danielstaleiny** [colemickens/nixpkgs-wayland#180]
Hi, thanks for this project <3 

Unstable-branch nixos fails to build sway dependency and rest fails.

```
sudo nixos-rebuild switch --show-trace
building Nix...
building the system configuration...
these derivations will be built:
  /nix/store/0dn97153kx0xhii4p12qdfsf2bcy5kr8-sway-1.2.drv
  /nix/store/xainkk6rcdhr2nwry5q3lngzxayn07a4-sway.drv
  /nix/store/aq4l7dhgdv6dydsjwclmrhafyj37wbl4-sway-joined.drv
  /nix/store/cxy0vrgcxgcz8r5njmvng8r0x1z0j20w-wayfire-1effe895f2209e78b5f0df59d30288e888e68710.drv
  /nix/store/g19qdnsvldaj2z2jwkx57p7834ja1flh-waybar-b9cd51a9cc9042166d94f6127ab37c60ff3e917f.drv
  /nix/store/nmrqpfgs4rvhh85q6hn926y2ahzcvbi3-redshift-7da875d34854a6a34612d5ce4bd8718c32bec804.drv
  /nix/store/gm1q284lsa2rawszhs0nqf86cwq7j0s5-system-path.drv
  /nix/store/0ibr8hfmar6rf5sayzhnyihkj15cvrf1-set-environment.drv
  /nix/store/2hdy9sbzl00lmb5fvl4kghr145qqwpxg-unit-polkit.service.drv
  /nix/store/wa32jvfs52g6gslar354xx5ha8n7958j-dbus-1.drv
  /nix/store/jps5f6xvgchfx7s0vy3yji867w1p4v6i-unit-dbus.service.drv
  /nix/store/sjjgxh53jz15ndvfymxgpypqk84ay8g2-unit-systemd-fsck-.service.drv
  /nix/store/1p8gaskzj898vn7k7rxg3q24ic2qs8is-system-units.drv
  /nix/store/fwhq22j90mpbyqdymlsrb1grqlw6b5vv-etc-profile.drv
  /nix/store/b3g8vr26fz0fk2mjfsy7g41maqwqpry6-unit-dbus.service.drv
  /nix/store/r78xgrxmrriq0yx945napxim5kw1gydk-user-units.drv
  /nix/store/b0kciyjdfqcg5j8js2bv4y1sqvgjjl6j-etc.drv
  /nix/store/14pipipqwj9g7p2mwpkx6nqlq8dr3973-nixos-system-box-20.03pre196836.1c40ee6fc44.drv
building '/nix/store/nmrqpfgs4rvhh85q6hn926y2ahzcvbi3-redshift-7da875d34854a6a34612d5ce4bd8718c32bec804.drv'...
building '/nix/store/0dn97153kx0xhii4p12qdfsf2bcy5kr8-sway-1.2.drv'...
building '/nix/store/cxy0vrgcxgcz8r5njmvng8r0x1z0j20w-wayfire-1effe895f2209e78b5f0df59d30288e888e68710.drv'...
unpacking sources
unpacking source archive /nix/store/2ya3vvsb3310hkr9pzfhr306imkb87hm-source
source root is source
patching sources
configuring
meson flags: --buildtype=plain         --libdir=/nix/store/brvl1j1kgfkqamwxf0bspxd7y4fx2cpi-wayfire-1effe895f2209e78b5f0df59d30288e888e68710/lib --libexecdir=/nix/store/brvl1j1kgfkqamwxf0bspxd7y4fx2cpi-wayfire-1effe895f2209e78b5f0df59d30288e888e68710/libexec         --bindir=/nix/store/brvl1j1kgfkqamwxf0bspxd7y4fx2cpi-wayfire-1effe895f2209e78b5f0df59d30288e888e68710/bin --sbindir=/nix/store/brvl1j1kgfkqamwxf0bspxd7y4fx2cpi-wayfire-1effe895f2209e78b5f0df59d30288e888e68710/sbin         --includedir=/nix/store/brvl1j1kgfkqamwxf0bspxd7y4fx2cpi-wayfire-1effe895f2209e78b5f0df59d30288e888e68710/include         --mandir=/nix/store/brvl1j1kgfkqamwxf0bspxd7y4fx2cpi-wayfire-1effe895f2209e78b5f0df59d30288e888e68710/share/man --infodir=/nix/store/brvl1j1kgfkqamwxf0bspxd7y4fx2cpi-wayfire-1effe895f2209e78b5f0df59d30288e888e68710/share/info         --localedir=/nix/store/brvl1j1kgfkqamwxf0bspxd7y4fx2cpi-wayfire-1effe895f2209e78b5f0df59d30288e888e68710/share/locale         -Dauto_features=enabled         -Dwrap_mode=nodownload         --prefix=/nix/store/brvl1j1kgfkqamwxf0bspxd7y4fx2cpi-wayfire-1effe895f2209e78b5f0df59d30288e888e68710
The Meson build system
Version: 0.51.2
Source dir: /build/source
Build dir: /build/source/build
Build type: native build
DEPRECATION: Duplicated values in array option is deprecated. This will become a hard error in the future.
DEPRECATION: Duplicated values in array option is deprecated. This will become a hard error in the future.
DEPRECATION: Duplicated values in array option is deprecated. This will become a hard error in the future.
Project name: wayfire
Project version: 0.3.1
C compiler for the host machine: /nix/store/7hzmz83nv8khpwsk858yaac7y3idh0a7-gcc-wrapper-8.3.0/bin/cc (gcc 8.3.0 "gcc (GCC) 8.3.0")
unpacking sources
unpacking source archive /nix/store/vc6xwsb9iyrqx79y03rnd0ja8yprnj7r-source
source root is source
patching sources
applying patch /nix/store/l4g7lb333j3bm1qsfx723ymaqhwy85mi-sway-config-no-nix-store-references.patch
patching file config.in
applying patch /nix/store/sdmqf08f6l6wmqapm0ibys0a0s82hk3h-load-configuration-from-etc.patch
patching file meson.build
Hunk #1 succeeded at 127 (offset -2 lines).
patching file sway/config.c
Hunk #1 succeeded at 359 (offset 42 lines).
configuring
C++ compiler for the host machine: /nix/store/7hzmz83nv8khpwsk858yaac7y3idh0a7-gcc-wrapper-8.3.0/bin/c++ (gcc 8.3.0 "g++ (GCC) 8.3.0")
DEPRECATION: Duplicated values in array option is deprecated. This will become a hard error in the future.
Build machine cpu family: x86_64
Build machine cpu: x86_64
meson flags: --buildtype=plain         --libdir=/nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2/lib --libexecdir=/nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2/libexec         --bindir=/nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2/bin --sbindir=/nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2/sbin         --includedir=/nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2/include         --mandir=/nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2/share/man --infodir=/nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2/share/info         --localedir=/nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2/share/locale         -Dauto_features=enabled         -Dwrap_mode=nodownload         --prefix=/nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2 -Ddefault-wallpaper=false -Dxwayland=enabled -Dgdk-pixbuf=enabled -Dtray=enabled -Dman-pages=enabled
Found pkg-config: /nix/store/gyh4j0wwlry6wx4rkpb6bzyic2b1wjqx-pkg-config-0.29.2/bin/pkg-config (0.29.2)
Run-time dependency wayland-server found: YES 1.17.0
Run-time dependency wayland-client found: YES 1.17.0
The Meson build system
Version: 0.51.2
Source dir: /build/source
Build dir: /build/source/build
Build type: native build
Run-time dependency wayland-cursor found: YES 1.17.0
Run-time dependency wayland-protocols found: YES 1.18
Run-time dependency cairo found: YES 1.16.0
DEPRECATION: Duplicated values in array option is deprecated. This will become a hard error in the future.
DEPRECATION: Duplicated values in array option is deprecated. This will become a hard error in the future.
DEPRECATION: Duplicated values in array option is deprecated. This will become a hard error in the future.
Project name: sway
Project version: 1.2
Run-time dependency libdrm found: YES 2.4.99
Run-time dependency egl found: YES 19.1.5
Run-time dependency glesv2 found: YES 19.1.5
Run-time dependency glm found: YES 0.9.8
Run-time dependency libinput found: YES 1.14.1
Run-time dependency pixman-1 found: YES 0.38.4
Run-time dependency threads found: YES
Run-time dependency xkbcommon found: YES 0.8.4
Run-time dependency wlroots found: YES 0.9.1
Run-time dependency wf-config found: YES 0.3
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency libinotify found: NO (tried pkgconfig and cmake)
Run-time dependency libjpeg found: YES 2.0.3
Run-time dependency libpng found: YES 1.6.37
Library execinfo found: NO
Fetching value of define "WLR_HAS_XWAYLAND" with dependency wlroots: 1
Fetching value of define "WLR_HAS_X11_BACKEND" with dependency wlroots: 1
WARNING: pkgconfig variable 'includedir' not defined for dependency glesv2.
C compiler for the host machine: /nix/store/7hzmz83nv8khpwsk858yaac7y3idh0a7-gcc-wrapper-8.3.0/bin/cc (gcc 8.3.0 "gcc (GCC) 8.3.0")
Has header "GLES3/gl32.h" : NO
Configuring config.h using configuration
DEPRECATION: Duplicated values in array option is deprecated. This will become a hard error in the future.
Build machine cpu family: x86_64
Build machine cpu: x86_64
Found pkg-config: /nix/store/gyh4j0wwlry6wx4rkpb6bzyic2b1wjqx-pkg-config-0.29.2/bin/pkg-config (0.29.2)
Program wayland-scanner found: YES (/nix/store/76klnvzfjri1p5387bxflx4zfqbx6nn0-wayland-1.17.0/bin/wayland-scanner)
Message:
----------------
wayfire 0.3.1

debug output: false
     imageio: true
      gles32: false
    xwayland: true
 x11-backend: true
graphics dbg: false
----------------

Build targets in project: 28
Found ninja-1.9.0 at /nix/store/n17bxm6v25d9ns74fv60668bg7ldm3aw-ninja-1.9.0/bin/ninja
Run-time dependency json-c found: YES 0.13.1
Run-time dependency libpcre found: YES 8.43
Run-time dependency wayland-server found: YES 1.17.0
Run-time dependency wayland-client found: YES 1.17.0
Run-time dependency wayland-cursor found: YES 1.17.0
Run-time dependency wayland-egl found: YES 18.1.0
Run-time dependency wayland-protocols found: YES 1.18
Run-time dependency xkbcommon found: YES 0.8.4
Run-time dependency cairo found: YES 1.16.0
Run-time dependency pango found: YES 1.43.0
Run-time dependency pangocairo found: YES 1.43.0
Run-time dependency gdk-pixbuf-2.0 found: YES 2.38.2
building
build flags: -j12 -l12
Run-time dependency pixman-1 found: YES 0.38.4
Run-time dependency libevdev found: YES 1.7.0
Run-time dependency libinput found: YES 1.14.1
Run-time dependency libsystemd found: YES 243
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency libelogind found: NO (tried pkgconfig and cmake)
Run-time dependency xcb found: YES 1.13.1
Library m found: YES
Library rt found: YES
Program git found: NO
Subproject directory not found and wlroots.wrap file not found
Subproject  subprojects/wlroots is buildable: NO (disabling)
Run-time dependency wlroots found: YES 0.9.1
Fetching value of define "WLR_HAS_XWAYLAND" with dependency wlroots: 1
Found pkg-config: /nix/store/gyh4j0wwlry6wx4rkpb6bzyic2b1wjqx-pkg-config-0.29.2/bin/pkg-config (0.29.2)
Build-time dependency scdoc found: YES 1.10.0
Program /nix/store/b77spvv39zdqn93c78hjbr1gclbadsxh-scdoc-1.10.0/bin/scdoc found: YES (/nix/store/b77spvv39zdqn93c78hjbr1gclbadsxh-scdoc-1.10.0/bin/scdoc)
Program sh found: YES (/nix/store/l6h4ya0wzb4b8mr0y58k2gh2nhfql4sn-bash-4.4-p23/bin/sh)
Compiler for C supports arguments -fmacro-prefix-map=/prefix/to/hide=: YES
Configuring config.h using configuration
Build-time dependency wayland-scanner found: YES 1.17.0
Program /nix/store/76klnvzfjri1p5387bxflx4zfqbx6nn0-wayland-1.17.0/bin/wayland-scanner found: YES (/nix/store/76klnvzfjri1p5387bxflx4zfqbx6nn0-wayland-1.17.0/bin/wayland-scanner)
Configuring config using configuration
Configuring 00-defaults using configuration
Message:
Features:
xwayland: true
gdk-pixbuf: true
systemd: true
elogind: false
tray: true
man-pages: true

Build targets in project: 36
Found ninja-1.9.0 at /nix/store/n17bxm6v25d9ns74fv60668bg7ldm3aw-ninja-1.9.0/bin/ninja
building
build flags: -j12 -l12
[51/157] Compiling C++ object 'src/25a6634@@wayfire@exe/main.cpp.o'.t.cpp.o'.pp.o'.able-v2-protocol.c.o'.
../src/main.cpp: In function 'int handle_config_updated(int, uint32_t, void*)':
../src/main.cpp:49:9: warning: ignoring return value of 'ssize_t read(int, void*, size_t)', declared with attribute warn_unused_result [-Wunused-result]
     read(fd, buf, INOT_BUF_SIZE);
     ~~~~^~~~~~~~~~~~~~~~~~~~~~~~
[67/261] Compiling C object 'sway/345c74e@@sway@exe/desktop_output.c.o'.mouse.c.o'.-protocol.c.o'.stable-v1-protocol.c.o'.g_output_unstable_v1_xml_client_h with a custom command.custom command.
FAILED: sway/345c74e@@sway@exe/desktop_output.c.o
/nix/store/7hzmz83nv8khpwsk858yaac7y3idh0a7-gcc-wrapper-8.3.0/bin/cc -Isway/345c74e@@sway@exe -Isway -I../sway -Iinclude -I../include -Iprotocols -I/nix/store/xdy3pw0hz1pxyvnjf7pv6hy9fwnlrvan-cairo-1.16.0-dev/include/cairo -I/nix/store/ffwk5kcb9ski6l81ln3xc23m3c7bsqzi-freetype-2.10.1-dev/include/freetype2 -I/nix/store/ffwk5kcb9ski6l81ln3xc23m3c7bsqzi-freetype-2.10.1-dev/include -I/nix/store/1q8lgpfwvj1hcibxwa7c1l66g2wg4fhf-json-c-0.13.1-dev/include/json-c -I/nix/store/qdhhbw967yd561mgvzx4n5bayngfmgc5-libevdev-1.7.0/include/libevdev-1.0/ -I/nix/store/3xpbz60nqlnakiqknj7qii5758zc9g0a-libinput-1.14.1-dev/include -I/nix/store/fhzmppvrzks9pkhiyjr9qg0qizdwx6qr-pango-1.43.0-dev/include/pango-1.0 -I/nix/store/sqvbi2ypbnldgxl7fsx0jzcsvl8y0047-glib-2.62.0-dev/include -I/nix/store/sqvbi2ypbnldgxl7fsx0jzcsvl8y0047-glib-2.62.0-dev/include/glib-2.0 -I/nix/store/s7rq6rzl851kv70s92zwrnbsw86rnkwj-glib-2.62.0/lib/glib-2.0/include -I/nix/store/6wjcdly1vp219g4sp8cc0igy4l386m0q-pcre-8.43-dev/include -I/nix/store/4fl608fn4wfijpwp9dyiq3jm94v5s1yn-pixman-0.38.4/include/pixman-1 -I/nix/store/76klnvzfjri1p5387bxflx4zfqbx6nn0-wayland-1.17.0/include -I/nix/store/kkpw8xpc5l16p09pbq9n3fyplqnqgv70-wlroots-2b04857343732b6aa7b8c4512387560f7f936aa9/include -I/nix/store/1njs7wpyhy75mrrd5n73fdznx57wg041-libxkbcommon-0.8.4-dev/include -I/nix/store/hcf2967wr5vqa53lry2ffvh5vvnqjxvq-libxcb-1.13.1-dev/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Werror -std=c11 -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wundef -Wvla '-DSYSCONFDIR="/etc"' '-DNIX_SYSCONFDIR="//nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2/etc"' '-DSWAY_VERSION="1.2"' -fmacro-prefix-map=../= -MD -MQ 'sway/345c74e@@sway@exe/desktop_output.c.o' -MF 'sway/345c74e@@sway@exe/desktop_output.c.o.d' -o 'sway/345c74e@@sway@exe/desktop_output.c.o' -c ../sway/desktop/output.c
../sway/desktop/output.c: In function 'output_has_opaque_overlay_layer_surface':
../sway/desktop/output.c:378:25: error: 'struct wlr_layer_surface_v1' has no member named 'layer'
     wlr_layer_surface_v1->layer != ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY) {
                         ^~
../sway/desktop/output.c: In function 'send_presented_iterator':
../sway/desktop/output.c:660:2: error: implicit declaration of function 'wlr_presentation_send_surface_presented'; did you mean 'wlr_presentation_feedback_send_presented'? [-Werror=implicit-function-declaration]
  wlr_presentation_send_surface_presented(server.presentation,
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  wlr_presentation_feedback_send_presented
cc1: all warnings being treated as errors
[70/261] Compiling C object 'sway/345c74e@@sway@exe/desktop_layer_shell.c.o'.
FAILED: sway/345c74e@@sway@exe/desktop_layer_shell.c.o
/nix/store/7hzmz83nv8khpwsk858yaac7y3idh0a7-gcc-wrapper-8.3.0/bin/cc -Isway/345c74e@@sway@exe -Isway -I../sway -Iinclude -I../include -Iprotocols -I/nix/store/xdy3pw0hz1pxyvnjf7pv6hy9fwnlrvan-cairo-1.16.0-dev/include/cairo -I/nix/store/ffwk5kcb9ski6l81ln3xc23m3c7bsqzi-freetype-2.10.1-dev/include/freetype2 -I/nix/store/ffwk5kcb9ski6l81ln3xc23m3c7bsqzi-freetype-2.10.1-dev/include -I/nix/store/1q8lgpfwvj1hcibxwa7c1l66g2wg4fhf-json-c-0.13.1-dev/include/json-c -I/nix/store/qdhhbw967yd561mgvzx4n5bayngfmgc5-libevdev-1.7.0/include/libevdev-1.0/ -I/nix/store/3xpbz60nqlnakiqknj7qii5758zc9g0a-libinput-1.14.1-dev/include -I/nix/store/fhzmppvrzks9pkhiyjr9qg0qizdwx6qr-pango-1.43.0-dev/include/pango-1.0 -I/nix/store/sqvbi2ypbnldgxl7fsx0jzcsvl8y0047-glib-2.62.0-dev/include -I/nix/store/sqvbi2ypbnldgxl7fsx0jzcsvl8y0047-glib-2.62.0-dev/include/glib-2.0 -I/nix/store/s7rq6rzl851kv70s92zwrnbsw86rnkwj-glib-2.62.0/lib/glib-2.0/include -I/nix/store/6wjcdly1vp219g4sp8cc0igy4l386m0q-pcre-8.43-dev/include -I/nix/store/4fl608fn4wfijpwp9dyiq3jm94v5s1yn-pixman-0.38.4/include/pixman-1 -I/nix/store/76klnvzfjri1p5387bxflx4zfqbx6nn0-wayland-1.17.0/include -I/nix/store/kkpw8xpc5l16p09pbq9n3fyplqnqgv70-wlroots-2b04857343732b6aa7b8c4512387560f7f936aa9/include -I/nix/store/1njs7wpyhy75mrrd5n73fdznx57wg041-libxkbcommon-0.8.4-dev/include -I/nix/store/hcf2967wr5vqa53lry2ffvh5vvnqjxvq-libxcb-1.13.1-dev/include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Werror -std=c11 -DWLR_USE_UNSTABLE -Wno-unused-parameter -Wno-unused-result -Wundef -Wvla '-DSYSCONFDIR="/etc"' '-DNIX_SYSCONFDIR="//nix/store/y78p1aq0qqll4fp450rnwsshdph76fs8-sway-1.2/etc"' '-DSWAY_VERSION="1.2"' -fmacro-prefix-map=../= -MD -MQ 'sway/345c74e@@sway@exe/desktop_layer_shell.c.o' -MF 'sway/345c74e@@sway@exe/desktop_layer_shell.c.o.d' -o 'sway/345c74e@@sway@exe/desktop_layer_shell.c.o' -c ../sway/desktop/layer_shell.c
In file included from ../sway/desktop/layer_shell.c:9:
../sway/desktop/layer_shell.c: In function 'handle_layer_shell_surface':
../sway/desktop/layer_shell.c:489:42: error: 'struct wlr_layer_surface_v1' has no member named 'layer'
   layer_surface->namespace, layer_surface->layer, layer_surface->layer,
                                          ^~
../include/log.h:44:62: note: in definition of macro 'sway_log'
  _sway_log(verb, "[%s:%d] " fmt, _SWAY_FILENAME, __LINE__, ##__VA_ARGS__)
                                                              ^~~~~~~~~~~
../sway/desktop/layer_shell.c:489:64: error: 'struct wlr_layer_surface_v1' has no member named 'layer'
   layer_surface->namespace, layer_surface->layer, layer_surface->layer,
                                                                ^~
../include/log.h:44:62: note: in definition of macro 'sway_log'
  _sway_log(verb, "[%s:%d] " fmt, _SWAY_FILENAME, __LINE__, ##__VA_ARGS__)
                                                              ^~~~~~~~~~~
../sway/desktop/layer_shell.c:546:46: error: 'struct wlr_layer_surface_v1' has no member named 'layer'
  wl_list_insert(&output->layers[layer_surface->layer], &sway_layer->link);
                                              ^~
[78/261] Compiling C object 'sway/345c74e@@sway@exe/desktop_render.c.o'.'.Kc.o'.
ninja: build stopped: subcommand failed.
builder for '/nix/store/0dn97153kx0xhii4p12qdfsf2bcy5kr8-sway-1.2.drv' failed with exit code 1
cannot build derivation '/nix/store/b0kciyjdfqcg5j8js2bv4y1sqvgjjl6j-etc.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/aq4l7dhgdv6dydsjwclmrhafyj37wbl4-sway-joined.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/g19qdnsvldaj2z2jwkx57p7834ja1flh-waybar-b9cd51a9cc9042166d94f6127ab37c60ff3e917f.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/14pipipqwj9g7p2mwpkx6nqlq8dr3973-nixos-system-box-20.03pre196836.1c40ee6fc44.drv': 1 dependencies couldn't be built
error: build of '/nix/store/14pipipqwj9g7p2mwpkx6nqlq8dr3973-nixos-system-box-20.03pre196836.1c40ee6fc44.drv' failed
```

Do you have suggestion to fix this ?? I can see that you fiddle with sway. in  commit https://github.com/colemickens/nixpkgs-wayland/commit/32afd2a70cbadfc214d353bc188efbeaedb3d4a9

**Comment by colemickens** [comment]
I'm not seeing any issues, and [the latest CI job]() is passing after fixing an unrelated issue. Additionally my system config builds as expected.

What channel are you on (and better, what commit). I assume `unstable` means `nixos-unstable` which is what this repo is built against (and what I personally use).

Finally, are you using `nixpkgs-wayland` as a package-set or overlay?

**Comment by danielstaleiny** [comment]
Yes, I am on nixos-unstable -> that is my only and default channel. 
```bash
nix-channel --list
nixos https://nixos.org/channels/nixos-unstable
```

I am using overlays 

I am not pinning the the your wayland package right now, just getting the master latest. 
```
....
  url = "https://github.com/colemickens/nixpkgs-wayland/archive/master.tar.gz";
  waylandOverlay = (import (builtins.fetchTarball url));
  schema = pkgs.gsettings-desktop-schemas;
  datadir = "${schema}/share/gsettings-schemas/${schema.name}";
in
{
  imports =
    [
      ./hardware-configuration.nix
    ];

  nixpkgs.overlays = [ waylandOverlay];
.....

```

**Comment by colemickens** [comment]
cc: @lovesegfault: Are you having any issues lately?

**Comment by colemickens** [comment]
Hi @danielstaleiny, I'm not sure what's up. The CI builds still look okay and I'm using it somewhat similar to how you're using it. Let's see if anyone else weighs in.

I assume you've updated the channel recently?

What should be happening -- `sway` should now come from nixpkgs, but it's just a wrapper and is NOT duplicated or provided by this overlay. An input to *that* `sway` wrapper is `sway-unwrapped` which **is** provided by this overlay now.

**Comment by colemickens** [comment]
Only other random idea - channels are managed by-user. Make sure that if you do `nix-channel --update` that you also do `sudo nix-channel --update`.

**Comment by danielstaleiny** [comment]
I would not have think that `nix-channel --update` without root could cause this issue.

I run updates on both root and user and rebuild and all  works now. Thank you very much.

**Comment by danielstaleiny** [comment]
Actually I have more information how it happened. I was able to reproduced the problem by playing with collecting garbage. 

So, as I previously said nix-channel --update on both root and user fix the issue and I was able to sudo rebuild system without the problem. 

after everything working , I run `nix-collect-garbage -d` on both root and user. 

after that I run `sudo nixos-rebuild switch --upgrade` and it failed as described up there. Couldn't build sway. 

This is interesting because I even specify --upgrade which based on this page should perform update channels and rebuild it in 1 command. 

Anyway, because it was the same issue, I run `nix-channel --update` on both root and my user 

after that I run `sudo nixos-rebuild switch --upgrade` command building everything correctly as it supposed to be. 


Conclusion, 

`nix-collect-garbage -d` is removing crutial paths for sway to build and I need to run nix-channel update to re-create those paths back. :))

I hope this will help somebody.
